cartridges/int_algolia_sfra/cartridge/static/default/js/utils/productTileUtils.js

```diff
@@ -0,0 +1,43 @@
+var lastRequestTime = '';
+
+
+// Listen for AJAX requests
+// If the request is an add to cart request, check if the last request was less than 4 seconds ago
+// If it was, abort the request 
+// Workaround to not modifying orginal cartridge code to fix issue with multiple add to cart requests because of adding productTile to PDP page
```

Comment:
Can you explain in more details from where this second comes from?

Reply:
one of them is coming from `cartridges/app_storefront_base/cartridge/client/default/js/product/detail..js` and other one is from `cartridges/app_storefront_base/cartridge/client/default/js/product/quickView.js`

Reply:
Ok, but how are they triggered? There are `onclick` listeners on the `.add-to-cart` button, but how can they be triggered multiple times?

Reply:
because we are registering listener in `storefront-reference-architecture/cartridges/app_storefront_base/cartridge/client/default/js/product/base.js` twice